File: Ejercicios Propuestos 1/Ejercicio 21/EJERCICIO_21.cpp
#include <iostream>

using namespace std;

int main() {
    int maximo_valor, numero_actual;
    int fila, columna;
    
    cout << "Ingrese el valor maximo hasta el que desea llegar: ";
    cin >> maximo_valor;
    
    if (maximo_valor > 0) {
        numero_actual = 1;
        fila = 1;
        while (numero_actual <= maximo_valor) {
            for (columna = 1; columna <= fila; columna++) {
                if (numero_actual <= maximo_valor) {
                    cout << numero_actual << " ";
                    numero_actual = numero_actual + 1;
                }
            }
            cout << endl;
            fila = fila + 1;
        }
    } else {
        cout << "Error: El valor debe ser mayor a cero." << endl;
    }
    
    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 21/Ejercicio 21.jl
print("Ingrese el valor maximo hasta el que desea llegar: ")
maximo_valor = parse(Int, readline())

if maximo_valor > 0
    numero_actual = 1
    fila = 1
    while numero_actual <= maximo_valor
        for columna in 1:fila
            if numero_actual <= maximo_valor
                print(numero_actual, " ")
                global numero_actual += 1
            end
        end
        println()
        global fila += 1
    end
else
    println("Error: El valor debe ser mayor a cero.")
end
